### ISSUE TITLE
OpenCL self-check mismatch? help?

### ISSUE THREAD

**Issue by wicatu** [gcp/leela-zero#1284]
Error in OpenCL calculation: expected -2.227215 got -41.186386 (error=1749.232674%)
Update your GPU drivers or reduce the amount of games played simultaneously.
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: OpenCL self-check mismatch.
Abort trap: 6

clCreateContextFromType(NULL, CL_DEVICE_TYPE_GPU)  Success (1)
    Platform Name                                 Apple
    Device Name                                   HD Graphics 4000
  clCreateContextFromType(NULL, CL_DEVICE_TYPE_ACCELERATOR)  No devices found in platform
  clCreateContextFromType(NULL, CL_DEVICE_TYPE_CUSTOM)  Invalid device type for platform
  clCreateContextFromType(NULL, CL_DEVICE_TYPE_ALL)  Success (2)
    Platform Name                                 Apple
    Device Name                                   HD Graphics 4000
    Device Name                                   Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
	NOTE:	your OpenCL library only supports OpenCL 1.0,

**Comment by Ttl** [comment]
Your OpenCL drivers are not working correctly. Use the CPU only version.

**Comment by wicatu** [comment]
how can  I start the CPU only version?

**Comment by wicatu** [comment]
sorry ,i am not good at it.i am confused by the following options.
./leelaz -h
Leela Zero 0.13  Copyright (C) 2017-2018  Gian-Carlo Pascutto and contributors
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; see the COPYING file for details.

Allowed options:
  -h [ --help ]                 Show commandline options.
  -g [ --gtp ]                  Enable GTP mode.
  -t [ --threads ] arg (=2)     Number of threads to use.
  -p [ --playouts ] arg         Weaken engine by limiting the number of 
                                playouts.Requires --noponder.
  -v [ --visits ] arg           Weaken engine by limiting the number of visits.
  --timemanage arg (=auto)      [auto|on|off] Enable extra time management 
                                features.
                                auto = off when using -m, otherwise on
  -b [ --lagbuffer ] arg (=100) Safety margin for time usage in centiseconds.
  -r [ --resignpct ] arg (=-1)  Resign when winrate is less than x%.
                                -1 uses 10% but scales for handicap.
  -m [ --randomcnt ] arg (=0)   Play more randomly the first x moves.
  -n [ --noise ]                Enable policy network randomization.
  -s [ --seed ] arg             Random number generation seed.
  -d [ --dumbpass ]             Don't use heuristics for smarter passing.
  -w [ --weights ] arg          File with network weights.
  -l [ --logfile ] arg          File to log input/output to.
  -q [ --quiet ]                Disable all diagnostic output.
  --noponder                    Disable thinking on opponent's time.
  --benchmark                   Test network and exit. Default args:
                                -v3200 --noponder -m0 -t1 -s1.
  --gpu arg                     ID of the OpenCL device(s) to use (disables 
                                autodetection).
  --full-tuner                  Try harder to find an optimal OpenCL tuning.
  --tune-only                   Tune OpenCL only and then exit.

**Comment by Ttl** [comment]
It's a separate download. If you are compiling yourself comment out "#define USE_OPENCL" in config.h.

**Comment by wicatu** [comment]
in sorrow
only drivers for windows. not drivers for mac.

**Comment by Splee99** [comment]
@wicatu On this webpage: https://support.apple.com/en-ca/HT202823, it states that the Intel HD 4000 should support Opencl 1.2, while what you have on your MAC book is Opencl 1.0. I think you should be able upgrade the driver. I'm not an expert on Mac, but I just guess you may find the update procedure on that website.

**Comment by iwestlin** [comment]
Hello, everyone, I have same issue on macbook air(early 2015).
This is my system info:  
![](https://ws1.sinaimg.cn/large/005CFza1gy1frjabb199xj306d04lq39.jpg)

According to https://support.apple.com/en-us/HT204349#intelhd, my air only has Intel HD Graphics 6000 inside. But when I ran `clinfo` this comes out:  
![](https://ws1.sinaimg.cn/large/005CFza1gy1frjavs7ld7j30i70acwgv.jpg)

I believe this is the problem which cause `mismatch` error, I can't remember if I installed any program somehow set this (fake) device up, and I'm wondering if anyone else have same output with clinfo on macbook air.

I googled a lot trying to delete this device for OpenCL, but got no lucky... 😭

So, is there any one who can teach me how to remove or disable the wrong device or change it to the right one? (which I believe is `Intel HD Graphics 6000`)
Many thanks. ❤️
